lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -135,6 +136,61 @@ def add_airline_data(post, options)
         post['order']['additionalInput']['airlineData'] = airline_data
       end
 
+      def add_lodging_data(post, options)
+        return unless lodging_options = options[:lodging_data]
+
+        lodging_data = {}
+
+        lodging_data['charges'] = add_charges(lodging_options)
+        lodging_data['checkInDate'] = lodging_options[:check_in_date] if lodging_options[:check_in_date]
+        lodging_data['checkOutDate'] = lodging_options[:check_out_date] if lodging_options[:check_out_date]
+        lodging_data['folioNumber'] = lodging_options[:folio_number] if lodging_options[:folio_number]
+        lodging_data['isConfirmedReservation'] = lodging_options[:is_confirmed_reservation] if lodging_options[:is_confirmed_reservation]
+        lodging_data['isFacilityFireSafetyConform'] = lodging_options[:is_facility_fire_safety_conform] if lodging_options[:is_facility_fire_safety_conform]
+        lodging_data['isNoShow'] = lodging_options[:is_no_show] if lodging_options[:is_no_show]
+        lodging_data['isPreferenceSmokingRoom'] = lodging_options[:is_preference_smoking_room] if lodging_options[:is_preference_smoking_room]
+        lodging_data['numberOfAdults'] = lodging_options[:number_of_adults] if lodging_options[:number_of_adults]
+        lodging_data['numberOfNights'] = lodging_options[:number_of_nights] if lodging_options[:number_of_nights]
+        lodging_data['numberOfRooms'] = lodging_options[:number_of_rooms] if lodging_options[:number_of_rooms]
+        lodging_data['programCode'] = lodging_options[:program_code] if lodging_options[:program_code]
+        lodging_data['propertyCustomerServicePhoneNumber'] = lodging_options[:property_customer_service_phone_number] if lodging_options[:property_customer_service_phone_number]
+        lodging_data['propertyPhoneNumber'] = lodging_options[:property_phone_number] if lodging_options[:property_phone_number]
+        lodging_data['renterName'] = lodging_options[:renter_name] if lodging_options[:renter_name]
+        lodging_data['rooms'] = add_rooms(lodging_options)
+
+        post['order']['additionalInput']['lodgingData'] = lodging_data
```

Comment:
Wow, who knew there were so many lodging options!

Reply:
This PR is a lesson on those hidden lodging fee's.